include/sycl/algorithm/equal.hpp

```diff
@@ -78,23 +78,20 @@ bool equal(ExecutionPolicy& exec, ForwardIt1 first1, ForwardIt1 last1,
   do {
     int passes = 0;
```

Comment:
I noticed some more cleaning should be done there. `passes` will always be 0 in the lamda below. Not sure if that was a quick hack to always go through `r.workitem_get_from(p, a1, a2);` or if this is not intentend. I would rather not change this behaviour in this PR.